rosidl_generator_c/src/u16string_functions.c

```diff
@@ -115,6 +127,24 @@ rosidl_generator_c__U16String__len(const uint16_t * value)
   return len;
 }
 
+bool
+rosidl_generator_c__U16String__resize(
+  rosidl_generator_c__U16String * str, size_t n)
+{
+  if (!str) {
+    return false;
+  }
+  uint16_t * data = realloc(str->data, (n + 1) * sizeof(uint16_t));
```

Comment:
Similar to the string implementation, consider guarding against `n == SIZE_MAX`:

https://github.com/ros2/rosidl/blob/4dce828f1a0b021d3e16b97e560259a9201be862/rosidl_generator_c/src/string_functions.c#L81-L84

Reply:
Good point: see d9e6fad36892174ad0b1940053dc7641acde5f45.